src/ORM/UnionList.php

```diff
@@ -0,0 +1,226 @@
+    {
+        return $this->count() > 0;
+    }
+
+    /**
+     * Iterate over each SS_List, one after the other.
+     *
+     * @return ArrayIterator
+     */
+    public function getIterator()
+    {
+        foreach ($this->lists as $i => $list) {
+            foreach ($list as $record) {
+                yield $record;
+            }
+        }
+    }
+
+    /**
+     * Get array of each record in each SS_List.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $result = array();
+        foreach ($this->lists as $i => $list) {
+            foreach ($list as $record) {
+                $result[] = $record;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Walks the list using the specified callback
+     *
+     * @param callable $callback
+     * @return $this
+     */
+    public function each($callback)
+    {
+        foreach ($this as $item) {
+            $callback($item);
+        }
+        return $this;
+    }
+
+    public function debug()
+    {
+        $val = "<h2>" . $this->class . "</h2><ul>";
+        foreach ($this->toNestedArray() as $item) {
+            $val .= "<li style=\"list-style-type: disc; margin-left: 20px\">" . Debug::text($item) . "</li>";
+        }
+        $val .= "</ul>";
+        return $val;
+    }
+
+    /**
+     * Return this list as an array and every object it as an sub array as well
+     *
+     * @return array
+     */
+    public function toNestedArray()
+    {
+        $result = array();
+        foreach ($this->lists as $i => $list) {
+            foreach ($list as $item) {
+                if (is_object($item)) {
+                    if (method_exists($item, 'toMap')) {
+                        $result[] = $item->toMap();
+                    } else {
+                        $result[] = (array) $item;
+                    }
+                } else {
+                    $result[] = $item;
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Get first item in the first SS_List.
+     *
+     * @return array
+     */
+    public function first()
+    {
+        $list = isset($this->lists[0]) ? $this->lists[0] : null;
+        return ($list) ? $list->first() : null;
+    }
+
+    /**
+     * Get last item in the last SS_List.
+     *
+     * @return array
+     */
+    public function last()
+    {
+        $list = end($this->lists);
+        return ($list) ? $list->last() : null;
+    }
+
+     /**
+     * Returns an array of a single field value for each item in each list.
+     *
+     * @param string $colName
+     * @return array
+     */
+    public function column($colName = 'ID')
+    {
+        $result = array();
+        foreach ($this->lists as $i => $list) {
+            $result = array_merge($result, $list->column($colName));
+        }
+        return $result;
+    }
+
+    public function map($keyfield = 'ID', $titlefield = 'Title')
+    {
+        throw new LogicException(
```

Comment:
Because if you you have two DataLists, one of say "Team" DataObject and one of "SiteTree" DataObject, the IDs aren't going to merge cleanly. Open to suggestions though on how this might work.